### ISSUE TITLE
Bug in IdentityServerAuthenticationService.SignIn: 'idsrv.session' cookie not set if I pass 'scheme' explicitly

### ISSUE THREAD

**Issue by maximpashuk** [IdentityServer/IdentityServer4#2154]
I think there is a bug in [async Task IdentityServerAuthenticationService.SignInAsync(...)](https://github.com/IdentityServer/IdentityServer4/blob/18f092180495c5725b55a61e74879eec82d7a7d7/src/IdentityServer4/Hosting/IdentityServerAuthenticationService.cs#L61) method

I call this method with "scheme" parameter explicitly set, but **user session cookie not set**.
In my case 'scheme' != options.DefaultAuthenticationScheme

I don't get why logic of ```.SingIn()``` methiod is conditional and in some cases user session not set.
By my opinion user session cookie should always be set without any conditions.

Anyway, if I pass 'scheme' explicitly, I expect session cookie set.

IdentityServerAuthenticationServer.SignInAsync decorates the standard [AuthenticationService.SignInAsync](https://github.com/aspnet/HttpAbstractions/blob/fbbbb16305f57025eca71964ac5d7c9f43c42150/src/Microsoft.AspNetCore.Authentication.Core/AuthenticationService.cs#L142)
For Liskov substitution principle be correct, decorated logic should not be far different from base logic.

I found a [commit](https://github.com/IdentityServer/IdentityServer4/commit/18f092180495c5725b55a61e74879eec82d7a7d7#diff-724f7aafe3a87aa182a3fffe7ab98d5dL48) related to #1477 there logic was changed.

**Comment by leastprivilege** [comment]
What scheme do you want to pass in? Something else than the default sign-in scheme?

**Comment by maximpashuk** [comment]
Actually, I already solved my problem.

My case was: trying to use DefaultAuthenticationScheme as "Bearer" and DefaultSigninScheme as "idsrv".
In IdentityServerAuthenticationService.SignInAsync authscheme must be equal to signinscheme.
So I create this issue.

But later I have found a workaround: DefaultScheme is "idsrv", controllers where "Bearer" should be used I decorated with Authorize(AuthenticationSchemes="Bearer").

Now all the things works fine, but actually I surprised that method IdentityServerAuthenticationService.SignInAsync requires both schemes (Authentication and SignIn) to be set 'idsrv'. I expect that SignInScheme should be enough.

Actually, workaround exists and this issue can be closed.
